src/odemis/driver/andorshrk.py

```diff
@@ -2218,11 +2218,16 @@ def _onPositionUpdate(self, pos):
         self._updateWavelengthList()
 
     def _updateWavelengthList(self):
+        """
+        Updates the wavelength list MD based on the current spectrograph position.
+        """
         npixels = self.resolution.value[0]
         pxs = self.pixelSize.value[0] * self.binning.value[0]
         wll = self._spectrograph.getPixelToWavelength(npixels, pxs)
-        md = {model.MD_WL_LIST: wll}
-        self._detector.updateMetadata(md)
+        if len(wll) == 0 and model.MD_WL_LIST in self._metadata.keys():
+            self._metadata.pop(model.MD_WL_LIST, None)  # remove WL list from MD if empty
```

Comment:
The `pop` trick is to not have to bother if the key happens to not be in the dict. So either you use `pop()`, and remove the `model.MD_WL_LIST in self._metadata.keys()`, or you leave that part, but simply write `del self._metadata[model.MD_WL_LIST]`. 
BTW, you can simply write `key in dict` instead of `key in dict.keys()`.